src/llfs/ioring_log_device.test.hpp

```diff
@@ -0,0 +1,196 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#pragma once
+#ifndef LLFS_IORINT_LOG_DEVICE_TEST_HPP
+#define LLFS_IORINT_LOG_DEVICE_TEST_HPP
+
+#include <llfs/int_types.hpp>
+#include <llfs/log_device.hpp>
+#include <llfs/slot_reader.hpp>
+
+#include <batteries/env.hpp>
+
+#include <cstdlib>
+#include <functional>
+#include <random>
+#include <vector>
+
+namespace llfs {
+
+template <typename T>
+T read_test_var(const char* name, T default_value)
+{
+  using value_type = decltype(default_value);
+  const value_type value = batt::getenv_as<value_type>(name).value_or(default_value);
+  std::cout << name << "=" << value << std::endl;
+  return value;
+}
+
+inline void run_log_device_benchmark(
+    const std::function<void(usize log_size, bool create,
```

Comment:
I think its good not to conflate "familiar" with "simple."  Yes, it could be re-coded into more familiar patterns (I think?  It depends on what a given reader is familiar with...), but it's not necessarily going to be simpler by any objective measure.  This construction is the result of breaking apart the micro-benchmark that was originally written against `IoRingLogDevice` to discover its performance issues, which then motivated the creation of this new impl.  The difference between the two benchmarks was code that was very awkwardly in the "middle" of the main test loop.  If its alright with you, I'd like to defer this change until the next PR, when I remove the old `IoRingLogDevice` for good.  Then there will be no more need to have shared code (and different code) for two micro-benchmarks, because there will only be one :-).

Reply:
That sounds good to me! I'm not even against  this pattern, more just trying to figure out if it's a pattern unfamiliar to me, or a pattern more generally  unfamiliar.